From 9ffa45fc1a3a1741078562b3667d99cb55b6ac3b Mon Sep 17 00:00:00 2001
From: Ankitprasad Sharma <ankitprasad.r.sharma@intel.com>
Date: Tue, 18 Nov 2014 15:23:33 +0530
Subject: [PATCH 1/4] SQUASHME! [VPG]: drm/i915: Fix for a memleak issue in
 native sync path

SQUASHME! - This patch should be squashed into the following existing patch:
Author: Tvrtko Ursulin <tvrtko.ursulin@intel.com>
Original Author: Satyanantha RamaGopal M <rama.gopal.m.satyanantha@intel.com>
Date:   Fri Dec 20 10:45:02 2013 +0000
Change-Id: I95ec32340e19b9e3a011cdab9c03c82c92c9af51
REVERTME [VPG]: drm/i915: Adding HW Native Sync support

There is a memory leak reported from the i915 Driver side across
suspend/resume cycles. The native sync timeline wasn't being destroyed
on suspend but was being recreated everytime on resume, loosing the
reference to original pointer.

Issue: GMINL-4109
Change-Id: I09a73e9761c1718d49dcb33f90fc28408f6a0bc6
Signed-off-by: Ankitprasad Sharma <ankitprasad.r.sharma@intel.com>
Signed-off-by: Akash Goel <akash.goel@intel.com>
---
 drivers/gpu/drm/i915/intel_sync.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_sync.c b/drivers/gpu/drm/i915/intel_sync.c
index 0a2a3ad..395f4f2 100644
--- a/drivers/gpu/drm/i915/intel_sync.c
+++ b/drivers/gpu/drm/i915/intel_sync.c
@@ -162,7 +162,12 @@ int i915_sync_timeline_create(struct drm_device *dev,
 				const char *name,
 				struct intel_engine_cs *ring)
 {
-	struct i915_sync_timeline *obj = (struct i915_sync_timeline *)
+	struct i915_sync_timeline *obj;
+
+	if (ring->timeline)
+		return 0;
+
+	obj = (struct i915_sync_timeline *)
 		sync_timeline_create(&i915_sync_timeline_ops,
 				     sizeof(struct i915_sync_timeline),
 				     name);
-- 
1.7.9.5

